docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,334 @@
+``gammapy download``, ``gammapy jupyter``), only `gammapy image`_ can be considered as 
+potentially needed in a data analysis process. It actually creates a counts image from an 
+event-list file and a image that serves as a reference geometry. Hence, we have a code 
+set-up in ``gammapy.scripts``, that we will not use it for the moment to expose the 
+high-level interface API, but to develop a very small set of specific command line tools 
+identified (i.e. perform long-time processing tasks, see *Command line tools* section below)
+
+We have a set of `Jupyter notebooks`_ as examples of tutorials and recipes demonstrating the
+use of Gammapy. These notebooks are continuously tested and are one of the pillars of the 
+user documentation. We could check most of the use cases are covered by the high-level 
+interface with the help of these notebooks. We will have to translate most of them to use the
+high-level interface, but we could also use them as a basis for experimental automated 
+workflows driven by parametrized notebooks executed with  `papermill`_. (see *Outlook* 
+section below) Moreover, some Python scripts have been added recently to perform `benchmarks`_
+of Gammapy, surely we could rewrite some of all of these benchmarks to use the high-level interface. 
+
+We also have some *high-level analysis* classes in the API that concatenate several atomic 
+actions and provide rough estimated results for more complex processes. (i.e. 
+`SpectrumAnalysisIACT`_, `LightCurveEstimator`_) These classes would serve as a basis to 
+design and prototype some of the tools of the high-level interface.
+
+
+Proposal
+========
+
+We will develop a high-level interface Python API which uses a params-values configuration file 
+defined by the user. This API would be used in Python scripts, notebooks or in IPython sessions
+to perform simple and most common IACT analysis.
+
+We see then two main options on how to use the High Level Interface API:
+
+- Within an IPython session or notebook, mostly dealing with a manager object to perform specific tasks in an **interactive** analysis process
+- In a Python script or notebook, declaring the orchestration of the tasks with a manager object for an **automatized** process
+
+This high-level interface API is similar to what it is done in `Fermipy`_ or HAP in HESS, including
+also the options to save and recover session states, as well as serialization of intermediate data
+products and logging. It is flexible enough to allow the user to work with the API at any stage of the
+analysis process, and not only from the start to the very end or in automatized process.
+
+**Use cases**
+
+The use cases covered are in the scope of a single analysis and model, not parametrized variations in a 
+multidimensional grid space of variables, and within a single region (e.g. 10 deg region with 5 sources).
+
+The main use cases for analysis to be covered are:
+
+- 3D map analysis
+- 2D map analysis
+- 1D spectrum analysis
+
```

Comment:
a LightCurve analysis is interesting also (important for the AGN Science Case, that fits perfectly in the pre-contidions of this simple analysis)

Reply:
Yes, I've added as a use case a *Light curve estimation* (i.e. the notebook that we already have about light curves) 

See also[ comment below on light curves](https://github.com/gammapy/gammapy/pull/2219/files#r309143215) where it is said the use case *Complex or memory eager processing on lightcurves* is not considered for the moment.